pkg/scheduler/eventhandlers.go

```diff
@@ -461,6 +461,9 @@ func nodeSchedulingPropertiesChange(newNode *v1.Node, oldNode *v1.Node) *framewo
 	if nodeConditionsChanged(newNode, oldNode) {
 		return &queue.NodeConditionChange
 	}
+	if nodeAnnotationsChanged(newNode, oldNode) {
```

Comment:
Through your new test, I'm wondering what if node annotation and resource is changed at the same time for example. In that case, plugins miss the node annotation change event. So probably nodeSchedulingPropertiesChange should return multiple events in that case.

But I think this change is kinda out of scope of this issue. This PR is gonna be too big if we include the change of nodeSchedulingPropertiesChange. Would you working on the fix of this bug in another PR?

Reply:
Yes, this situation is indeed likely to occur. 

If that's the case, we should also pass events to [SchedulingQueue.MoveAllToActiveOrBackoffQueue](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/internal/queue/scheduling_queue.go#L764) since it currently processes only one event at a time. 

I'll submit a new PR to fix it.

Reply:
Thanks, could you move the change for `TestNodeSchedulingPropertiesChange` from this PR to that new PR?

Reply:
> Thanks, could you move the change for `TestNodeSchedulingPropertiesChange` from this PR to that new PR?

👌

Reply:
Can you do it in a preparatory PR that we can merge before this PR?

Reply:
> Can you do it in a preparatory PR that we can merge before this PR?

I've already submitted https://github.com/kubernetes/kubernetes/pull/118490 to address this issue.